ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -663,7 +663,7 @@ CompletableFuture<RaftClientReply> submitStepDownRequestAsync(TransferLeadership
     return pendingStepDown.submitAsync(request);
   }
 
-  private synchronized String sendStartLeaderElection(RaftPeerId follower, TermIndex lastEntry) {
+  synchronized String sendStartLeaderElection(RaftPeerId follower, TermIndex lastEntry) {
```

Comment:
`synchronized` is incorrect since it does not change anything in `LeaderStateImpl`.

Actually, this method has nothing to do with `LeaderStateImpl`.  It should be moved to `RaftServerImpl` since it use many methods there.

Reply:
Good catch. Thanks for the suggestion.